### PR TITLE
Update logic for dataset and dataset version

### PR DESCRIPTION
**Pull request by SophieGuo410**
This pr include the following changes to support friendly usage and fix bugs:
1. Change SQLException to AccountServiceException in order to make the error message more clear.
2. Return the latest expirationTime depends on Dataset level expirationTime and update blob properties.
dataset version ttl = Min(dataset version ttl, dataset ttl)
3. Add USER_META_DATA_HEADER_PREFIX for user tags so when people set user tag, they don't need to provide the prefix.
4. change userMetadataFromRequest to modifiable map in order to add userTags.
5. add more test to coverage the changes.

### CODE REVIEW

File: ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java [hunk 1]
```diff
@@ -390,44 +394,27 @@ public synchronized DatasetVersionRecord getDatasetVersions(short accountId, sho
...
     statement.setString(3, datasetName);
     statement.setLong(4, version);
-    if (expirationTimeMs != Infinite_Time) {
-      statement.setTimestamp(5, new Timestamp(expirationTimeMs));
+    long updatedExpirationTimeMs = expirationTimeMs;
+    if (expirationTimeMs == Infinite_Time && dataset.getExpirationTimeMs() == Infinite_Time) {
+      statement.setTimestamp(5, null);
+    } else if (expirationTimeMs == Infinite_Time || dataset.getExpirationTimeMs() == Infinite_Time) {
+      updatedExpirationTimeMs = expirationTimeMs == Infinite_Time ? dataset.getExpirationTimeMs() : expirationTimeMs;
+      statement.setTimestamp(5, expirationTimeMs == Infinite_Time ? new Timestamp(dataset.getExpirationTimeMs())
```
Comment (justinlin-linkedin) [inline ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java:408]: `statement.setTimestamp(5, new Timestamp(updatedExpirationTimeMs));`

File: ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java [hunk 2]
```diff
@@ -390,44 +394,27 @@ public synchronized DatasetVersionRecord getDatasetVersions(short accountId, sho
...
+    } else if (dataset.getExpirationTimeMs() <= expirationTimeMs) {
+      updatedExpirationTimeMs = dataset.getExpirationTimeMs();
+      statement.setTimestamp(5, new Timestamp(dataset.getExpirationTimeMs()));
     } else {
-      if (dataset.getExpirationTimeMs() != Infinite_Time) {
-        statement.setTimestamp(5, new Timestamp(dataset.getExpirationTimeMs()));
-      } else {
-        statement.setTimestamp(5, null);
-      }
+      statement.setTimestamp(5, new Timestamp(expirationTimeMs));
```
Comment (justinlin-linkedin) [inline ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java:414]:
```
} else {
    updatedExpirationTimeMs = Math.min(dataset.getExpirationTimeMs(), expirationTimeMs);
    statement.setTimestamp(5, new Timestamp(updatedExpirationTimeMs));
}
```

File: ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java (hunk 1 above)
Comment (SophieGuo410) [reply ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java]: Updated.

File: ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java (hunk 2 above)
Comment (SophieGuo410) [reply ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java]: Updated.

**Review by justinlin-linkedin — APPROVED** [review]

**Review by mikeyang923 — APPROVED** [review]
lgtm

**Pull request merged**